rclpy/threading/data_plotter.py

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env python3
+
+# Copyright 2016 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from copy import deepcopy
+from threading import Lock, Thread
+import time
+
+import rclpy
+
+import matplotlib.pyplot as plt
+
+from std_msgs.msg import Int64
+
+
+# This script starts a node with a subscription to a relatively high-frequency topic.
+# The data received from the subscription callbacks is plotted in the main thread,
+# while the callbacks themselves are processed in a second thread.
+# Plot updates are scheduled at a regular interval, independent of the rate at which data is
+# received. Slow plotting calls will not block the reception of frequent data messages.
+
+time_between_plot_updates = 1  # time in seconds between plot updates
+
+
+# Class for managing the storage and display of data received
+class DataPlotter:
+    def __init__(self):
+        plt.ion()
+        self.fig = plt.figure()
+        self.received_data = []
+        # As the received data can be access by both the main plotting thread and the thread
+        # processing data callbacks, its access must be protected by a lock
+        self.received_data_lock = Lock()
+
+    # Called when new data is received
+    def data_callback(self, msg):
+        print('Data received: {0}'.format(msg.data))
```

Comment:
I personally prefer the `.format()`, but either is fine. I'm not sure if we ever decided to only do one or the other.